packages/hoppscotch-app/components/http/Parameters.vue

```diff
@@ -204,32 +203,40 @@ useCodemirror(bulkEditor, bulkParams, {
 
 const params$ = useReadonlyStream(restParams$, [])
 
-watch(
-  params$,
-  (newValue) => {
-    if (
-      (newValue[newValue.length - 1]?.key !== "" ||
-        newValue[newValue.length - 1]?.value !== "") &&
-      newValue.length
+const editBulkParamLine = (index: number, item?: HoppRESTParam) => {
+  const params = bulkParams.value.split("\n")
+
+  if (item !== null)
+    params.splice(
+      index,
+      1,
+      `${!item.active ? "// " : ""}${item.key}: ${item.value}`
     )
-      addParam()
-  },
-  { deep: true }
-)
+  else params.splice(index, 1)
+
+  bulkParams.value = params.join("\n")
+}
```

Comment:
Hi @0xc0Der, this doesn't ignore the empty fields in the list and sets the last entry which has a key-value pair.

I guess we should keep the flexibility to keep empty entries in key-value pairs as well as empty lines in bulk editors since timing those is causing Cumulative Layout Shift (CLS) throughout the UI.

What do you think?
Everything else looks good though.

Reply:
so, the empty fields you added at the end which went away when you started editing is an expected behavior considering the structure of the code. then the field you added got appended at the end.
did i get that right?
keeping empty fields at the end is not important in my opinion _we have an "add" button_ but the empty fields in the middle are preserved as you see.
am i off point here? please correct me if i am wrong.

Reply:
if you want to preserve these empty fields, i am working on it.
another problem to think about:
bulk editor contents does not get saved in local storage, then it does not get set at startup.